ide_extensions.md

```diff
@@ -0,0 +1,161 @@
+# IDE Extensions
+## Overview
+An oft-requested feature is an improved `Tiltfile` editing experience in IDEs/editors.
+
+Specifically, for the scope of this spec, we'll consider two high-level aspects:
+ * Syntax highlighting
+ * Advanced IDE features (autocomplete, go to definition, etc.)
```

Comment:
i think i would separate this out into two bits:
- autocomplete docs for built-ins
- autocomplete + navigation for user-defined code

in practice I think the solution spaces of those two problems are going to be very different. The first one is what we care about more for people writing new tiltfiles.

Reply:
i asked Joey about https://play.authzed.com/schema. he said they have no LSP support right now. they simply pre-generate a schema file. Then the autocompleter tokenizes the sample schema and provides autocomplete suggestion based on the existing tokenset LOL.

so maybe we can cheat a bit in places : )

Reply:
We can definitely cheat a lot!! What you're describing is very similar to the `vscode-anycode` extension approach - it uses tree-sitter to parse the file, then provides context-unaware auto-complete using the symbols it found.